### DISCUSSION TITLE
The GPT-4 API works worse than Chat GPT-4.

### DISCUSSION THREAD

**Discussion by antonleonov1983** [n3d1117/chatgpt-telegram-bot#303] [General]
GPT-4 API answers questions less accurately and in-depth.
GPT-4 API, when asked "What is your engine version?" - answers that it's GPT-3.
Chat GPT-4, when asked "What is your engine version?" - answers that it's GPT-4.
Are GPT-4 API and Chat GPT-4 on different engines? Or can settings be adjusted to achieve the same efficiency?

**Comment by Shady-Dev** [comment]
Did you enable gpt-4 in the .env file?

**Reply by fpwex9** [reply]
Hello! I am having the same issue. 

I have `OPENAI_MODEL=gpt-4` set in the .env file. 
When I ask it the bot:
> Which GPT model do you use?

the response is:
> As an AI language model created by OpenAI, I am powered by GPT-3, the third iteration of the Generative Pre-trained Transformer technology. GPT-3 is known for its large-scale capacity and advanced natural language processing capabilities.

Whereas web based ChatGPT-4 responding to the same query: 
> I am based on the GPT-4 architecture developed by OpenAI. This is a continuation of the development of GPT-3 with additional improvements and updates for more accurate and natural interaction with users.

**Reply by tive-tilray** [reply]
Hey! You're correct, the GPT model doesn't know which version it is until it's told via a System Message. This is how developers can configure the model to be aware of which version it represents.

As for token usage, that's also a good way to check which version of the model you're using. GPT-4 and GPT-3 have different token limits and different token usages for various tasks. So, if you notice that the token usage is different from what you'd expect, that could be a clue as to which version of the model you're using.

**Comment by antonleonov1983** [comment]
I also noticed that in addition to the lower level of intelligence, the GPT 4 API sometimes puts incorrect word endings in other languages.
GPT 4 chat on the site, never mixed up the endings of words.
Probably different builds, what do you think?

**Comment by GuillaumeMoscou** [comment]
I have the **same behavior as described above** thought I received an email from Open AI saying that :
"Your OpenAI organization (xxxxx) can now access GPT-4 models with 8k context size via the existing OpenAI API"

How to properly install GPT 4 ? 

In my .env file this line is in place :
`OPENAI_MODEL="gpt-4" # or "gpt-4-0314" or "gpt-4-8k" or "gpt-4-32k-0314" or "gpt-4-0314" or "gpt-4-0613"`

**Comment by tive-tilray** [comment]
If you're using the GPT model and want to know which version you're interacting with, it's crucial to set this information in the System Message. The model itself won't know which version it is unless explicitly told by the developers through the System Message configuration.

Additionally, proper System Message settings and other configurations are key to how your chatbot behaves. Incorrect or incomplete settings can lead to unexpected behavior, so make sure to configure your chatbot carefully.

You can also check the token usage to get an idea of which version you're using. Different versions of GPT (GPT-3, GPT-4, etc.) have different token limits and usage patterns. Observing how many tokens are being used can give you a clue about the version.

**Reply by Jake-double-one** [reply]
The original ChatGPT prompts got leaked: https://github.com/LouisShark/chatgpt_system_prompt
For ChatGPT (v4) see here: https://github.com/LouisShark/chatgpt_system_prompt/blob/main/prompts/gpt-4-gizmo-20231116.md

Part of the initial chat prompt for ChatGPT is:
```
You are a ChatGPT, a large language model trained by OpenAI, based on the GPT-4 architecture.
Knowledge cutoff: 2023-04
Current date: 2023-11-16
```
That's why ChatGPT knows its architecture ;-)
